purchase_order_no_zero_price/models/purchase_order_line.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (C) 2022 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.constrains("price_unit", "display_type", "state")
+    def _check_price_unit_zero(self):
+        # Prevent zero prices on confirmed POs
+        # to help ensure proper Stock Valuation
+        # when receiving the products
+        lines = self.filtered(
+            lambda x: not x.price_unit
+            and x.discount != 100.0  # Allow zero proce for 100% discounts
```

Comment:
you could write a getattr(x, "discount", False). That way : 
- you don't have to set a dependency to ``purchase_discount``.
- you could make the module working with ``purchase_triple_discount`` writing similar code for ``discount2`` and ``discount3``.
-

Reply:
This has actually been deployed along with `purchase_triple_discount`, and that's not a bad idea.

Reply:
@legalsylvain Done, please review.